File: src/leetcode/java/easy/ContainsDuplicateII_219.java
package leetcode.java.easy;

import java.util.HashSet;

public class ContainsDuplicateII_219 {
  public boolean containsNearbyDuplicate(int[] nums, int k) {
    HashSet<Integer> set = new HashSet<>();
    for (int i = 0; i < nums.length; i++) {
      if (i > k) set.remove(nums[i - k - 1]);
      if (!set.add(nums[i])) return true;
    }
    return false;
  }
}

File: src/leetcode/python/ContainsDuplicateII_219.py
from collections import deque


class ContainsDuplicateII_219(object):
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        s = set()
        for i, v in enumerate(nums):
            if i > k:
                s.remove(nums[i - k - 1])
            if v in s:
                return True
            s.add(v)
        return False
